### PR TITLE
feat: quiz page

### PR DESCRIPTION
**Pull request by vrx29**
Live Preview : [Gimesto Quiz App Live](https://gimesto-quiz-git-quiz-vrx29.vercel.app/)

This PR contains functionality for displaying quiz questions and allowing user to select options and also navigate to next question.

### CODE REVIEW

File: src/pages/Quiz/index.tsx [hunk 1]
```diff
@@ -1,6 +1,103 @@
...
+  }, [quesNo, setQues, currentQuizQuestions]);
+
+  const handleOptionChange = (option) => {
+    setCurrentQuizQuestions(
+      currentQuizQuestions.map((q) => {
+        if (q._id === ques._id) {
+          return { ...q, isSelected: option };
+        }
+        return { ...q };
+      })
```
Comment (foolhardy21) [inline src/pages/Quiz/index.tsx:46]: suggestion - this can be converted to single line using `? : ` operator.

**Review by foolhardy21 — APPROVED** [review]
code looks good. use of url param for each question is good use of `useParams`. nice work.

File: src/components/QuizCard/index.tsx [hunk 2]
```diff
@@ -4,19 +4,21 @@ import { PlayIcon } from "assets/icons";
 import { Link } from "react-router-dom";
 
 export function QuizCard({ quiz }) {
-  const { difficulty, name, genres, poster } = quiz;
+  const { _id: quizId, difficulty, name, genres, poster } = quiz;
   return (
     <div className="category-card">
       <img src={poster} alt="" />
```
Comment (BalajiHarinath) [inline src/components/QuizCard/index.tsx]: Missing alt attribute

File: src/pages/Quiz/index.tsx (hunk 1 above)
Comment (vrx29) [reply src/pages/Quiz/index.tsx:46]: In next PR might be doing some more operations thats why did it like this.

File: src/pages/Quiz/quiz.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.options-cont label:hover {
+  background-color: var(--COLOR-OPTION-HOVER);
+  transform: translate(0.2rem);
+  transition: transform ease-in-out 0.5s;
+}
+
+.options-cont input[type="radio"]:checked + label {
+  background-color: var(--COLOR-PRIMARY);
```
Comment (BalajiHarinath) [inline src/pages/Quiz/quiz.css]: can remove the commented code

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggestions everything looks good.

File: src/pages/Quiz/quiz.css (hunk 3 above)
Comment (vrx29) [reply src/pages/Quiz/quiz.css]: done

File: src/components/QuizCard/index.tsx (hunk 2 above)
Comment (vrx29) [reply src/components/QuizCard/index.tsx]: done

**Pull request merged**